### PR TITLE
test: add e2e tests for consentRequests

### PR DESCRIPTION
**Pull request by sridharvoruganti**
Closes #[2001](https://github.com/mojaloop/project/issues/2001)

### CODE REVIEW

File: docs/linking/2-request-consent-otp.puml [hunk 1]
```diff
@@ -111,4 +119,76 @@ note over PISP, DFSP
...
+Switch --> DFSP: ""200 OK""
+deactivate DFSP
+
+Switch -> PISP ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
```
Comment (eoln) [inline docs/linking/2-request-consent-otp.puml:189]: NEW ERROR CASE: `authChannels` is an array that can have only one value. So we have an error case when specified channel isn't allowed for given userId

File: docs/linking/2-request-consent-web.puml [hunk 2]
```diff
@@ -137,4 +141,73 @@ DFSPAuthorizeSimulator -> DFSPAuthorizeSimulator: generate a **secret="some-secr
...
+Switch --> DFSP: ""200 OK""
+deactivate DFSP
+
+Switch -> PISP ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
```
Comment (eoln) [inline docs/linking/2-request-consent-web.puml:208]: NEW ERROR CASE: `authChannels` is an array that can have only one value. So we have an error case when specified channel isn't allowed for given userId

**Review by eoln — DISMISSED** [review]

File: docker-local/dfsp_b/rules.json [hunk 3]
```diff
@@ -56,17 +56,124 @@
         "statusCode": 200,
         "body": [
           {
-            "accountNickname": "dfspa.user.nickname1",
-            "id": "dfspa.username.1234",
+            "accountNickname": "dfspb.user.nickname1",
```
[the remark this answers is not in the document]
Comment (lewisdaly) [reply docker-local/dfsp_b/rules.json:59]: I don't think this is an issue- the PISP should just pass on whatever currency it sees. The demo app does need some tweaking however...

File: docs/linking/2-request-consent-otp.puml (hunk 1 above)
Comment (lewisdaly) [reply docs/linking/2-request-consent-otp.puml:189]: for `POST /consentRequests` this must be an array with 2 values, so not really an issue... if the dfsp rejects the `POST /consentRequests`, that can be considered a generic rejection, right?

File: docker-local/dfsp_a/rules.json [hunk 4]
```diff
@@ -95,5 +95,112 @@
...
+    "event": {
+      "type": "FIXED_CALLBACK",
+      "params": {
+        "statusCode": 200,
+        "body": {
+          "isValid": true,
+          "authChannels": [
+            "WEB"
+          ],
+          "authUri": "dfspa.com/authorize?consentRequestId=xxx-xxxx"
```
[the remark this answers is not in the document]
Comment (eoln) [reply docker-local/dfsp_a/rules.json:131]: if the simulator will propagate `consentRequestId` in `authUri` than we can use it also for some clickable demo to show redirecting to `dfsp login page` - am I right?

File: docs/linking/2-request-consent-web.puml [hunk 5]
```diff
@@ -137,4 +141,73 @@ DFSPAuthorizeSimulator -> DFSPAuthorizeSimulator: generate a **secret="some-secr
 DFSPAuthorizeSimulator -> WebBrowser: 201 OK: response: { **secret:"some-secret"**" }
 WebBrowser -> User: redirects to callbackUri="pisp-app://callback...?**secret=some-secret**"
 deactivate User
+
+== Failure case ==
```
Comment (lewisdaly) [inline docs/linking/2-request-consent-web.puml:145]: Please add as a standalone puml

File: docs/linking/2-request-consent-otp.puml [hunk 6]
```diff
@@ -111,4 +119,76 @@ note over PISP, DFSP
   At this point, the PISP knows that the OTP authChannel is in use and the PISP App should prompt the user to provide the OTP.
 end note
 
+== Failure case ==
```
Comment (lewisdaly) [inline docs/linking/2-request-consent-otp.puml:122]: Please add as a standalone puml

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: docker-local/dfsp_a/rules.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply docker-local/dfsp_a/rules.json]: Ok now returning request consentRequestId. yes we can use that id in demo auth\login page.

File: docs/linking/2-request-consent-web.puml (hunk 5 above)
Comment (sridharvoruganti) [reply docs/linking/2-request-consent-web.puml]: Ok done

File: docs/linking/2-request-consent-otp.puml (hunk 6 above)
Comment (sridharvoruganti) [reply docs/linking/2-request-consent-otp.puml]: Ok

File: docs/linking/1-discovery.puml [hunk 7]
```diff
@@ -21,8 +21,6 @@ activate PISP
```
Comment (eoln) [inline docs/linking/1-discovery.puml:21]: @lewisdaly `AccountsIDPutResponse` response shouldn't be an array but object with `accounts` property 

In api-snippets is the old (end potentially dangerous) version: https://github.com/mojaloop/api-snippets/blob/master/thirdparty/openapi3/components/schemas/AccountsIDPutResponse.yaml

File: docs/linking/README.md [hunk 8]
```diff
@@ -375,6 +379,25 @@ TODO!
 
 # 4. Linking Error Scenarios
 
+## 4.1. Discovery
+![Accounts error](../out/linking/error_scenarios/1-discovery-error.svg)
+
+## 4.2. Bad consentRequests
+When the DFSP receives the POST /consentRequests request from the PISP, any number of processing or validation errors could occur, such as:
+
+1. FSP does not support any requested authentication channels
```
Comment (eoln) [inline docs/linking/README.md:388]: I would point out here error codes values for each error-case

**Review by eoln — APPROVED** [review]
That is really great we have finally documented error cases!
I am really happy to see superb high-quality work done by @sridharvoruganti 
- little improvement in doc can be done -> let put on the list of error cases also error codes

File: docker-local/dfsp_a/rules.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (eoln) [reply docker-local/dfsp_a/rules.json]: great @sridharvoruganti :-)

File: docs/linking/1-discovery.puml (hunk 7 above)
Comment (lewisdaly) [reply docs/linking/1-discovery.puml:21]: Ok, yeah we need to fix this in the api draft and our sequences. I'll make a note to follow it up.

File: docs/linking/error_scenarios/1-discovery-error.puml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  ""  } ""\n\
+  ""}""
+PISP --> Switch: ""200 OK""
+deactivate Switch
+deactivate PISP
+
+...
+
+note over App, DFSP
+  The PISP can now show error message and user can try again with another username.
```
Comment (lewisdaly) [inline docs/linking/error_scenarios/1-discovery-error.puml:62]:
```suggestion
  The PISP can now show error message and user can try again with another username, or different DFSP.
```

**Review by lewisdaly — APPROVED** [review]

File: docs/linking/README.md (hunk 8 above)
Comment (sridharvoruganti) [reply docs/linking/README.md]: Ok error codes added

File: docs/linking/error_scenarios/1-discovery-error.puml (hunk 9 above)
Comment (sridharvoruganti) [reply docs/linking/error_scenarios/1-discovery-error.puml]: Ok

**Review by eoln — APPROVED** [review]
lgtm

**Pull request merged**